### PR TITLE
Consumer panic cleanup

### PR DESCRIPTION
**Pull request by mpoke**
# Description

Audit of the panic calls in `x/consumer`

## Linked issues

Works toward https://github.com/cosmos/interchain-security/issues/621

## Type of change

- [x] `Refactor`: Changes existing code style, naming, structure, etc.
- [x] `Docs`: Adds documentation

## Regression tests

The existing UTs are covering the methods modified.

### CODE REVIEW

File: x/ccv/consumer/keeper/relay.go [hunk 1]
```diff
@@ -99,9 +96,6 @@ func (k Keeper) OnRecvVSCPacket(ctx sdk.Context, packet channeltypes.Packet, new
 // the consumer chain.
 func (k Keeper) QueueVSCMaturedPackets(ctx sdk.Context) {
 	for _, maturityTime := range k.GetElapsedPacketMaturityTimes(ctx) {
-		if ctx.BlockTime().Before(maturityTime.MaturityTime) {
```
Comment (mpoke) [inline x/ccv/consumer/keeper/relay.go:102]: Removed. Redundant check.

File: x/ccv/utils/utils.go [hunk 2]
```diff
@@ -44,15 +44,6 @@ func AccumulateChanges(currentChanges, newChanges []abci.ValidatorUpdate) []abci
 	return out
 }
 
-func GetChangePubKeyAddress(change abci.ValidatorUpdate) (addr []byte) {
```
Comment (mpoke) [inline x/ccv/utils/utils.go:47]: Removed as it was called only from `ApplyCCValidatorChanges`. Added the logic directly there.

File: x/ccv/consumer/keeper/validators.go [hunk 3]
```diff
@@ -17,7 +16,14 @@ import (
...
 	for _, change := range changes {
-		addr := utils.GetChangePubKeyAddress(change)
+		// convert TM pubkey to SDK pubkey
+		pubkey, err := cryptocodec.FromTmProtoPublicKey(change.GetPubKey())
+		if err != nil {
+			// An error here would indicate that the validator updates
+			// received from the provider are invalid.
+			panic(err)
+		}
+		addr := pubkey.Address()
```
Comment (mpoke) [inline x/ccv/consumer/keeper/validators.go:26]: Moved here the logic from `utils.GetChangePubKeyAddress`.

**Review by danwt — APPROVED** [review]
Looks fine 👍

**Review by smarshall-spitzbart — APPROVED** [review]
Nice!

**Pull request merged**